model/auth.go

```diff
@@ -90,74 +79,73 @@ func (a *Auth) UnmarshalJSON(data []byte) error {
 	if len(a.Scheme) == 0 {
 		a.Scheme = AuthTypeBasic
 	}
-	authProperties, ok := getAuthProperties(a.Scheme)
-	if !ok {
-		return fmt.Errorf("authentication scheme %s not supported", a.Scheme)
-	}
 
-	// we take the type we want to unmarshal based on the scheme
-	if err := unmarshalKey("properties", auth, authProperties); err != nil {
-		return err
-	}
+	switch a.Scheme {
+	case AuthTypeBasic:
+		authProperties := &BasicAuthProperties{}
 
-	a.Properties = authProperties
-	return nil
-}
+		if err := unmarshalKey("properties", auth, authProperties); err != nil {
+			return err
+		}
+		a.Properties.Basic = authProperties
 
-// AuthProperties ...
-type AuthProperties interface {
-	// GetMetadata ...
-	GetMetadata() *Metadata
-	// GetSecret ...
-	GetSecret() string
-	// DeepCopyAuthProperties fixes in.Properties.DeepCopyAuthProperties undefined (type AuthProperties has no
-	// field or method DeepCopyAuthProperties)
-	DeepCopyAuthProperties() AuthProperties
-}
+		return nil
 
-// BaseAuthProperties ...
-type BaseAuthProperties struct {
-	Common `json:",inline"`
-	// Secret Expression referencing a workflow secret that contains all needed auth info
-	Secret string `json:"secret,omitempty"`
-}
+	case AuthTypeBearer:
+		authProperties := &BearerAuthProperties{}
+		if err := unmarshalKey("properties", auth, authProperties); err != nil {
+			return err
+		}
+		a.Properties.Bearer = authProperties
+		return nil
 
-// UnmarshalJSON ...
-func (b *BaseAuthProperties) UnmarshalJSON(data []byte) error {
-	properties := make(map[string]json.RawMessage)
-	if err := json.Unmarshal(data, &properties); err != nil {
-		b.Secret, err = unmarshalString(data)
-		if err != nil {
+	case AuthTypeOAuth2:
+		authProperties := &OAuth2AuthProperties{}
+		if err := unmarshalKey("properties", auth, authProperties); err != nil {
 			return err
 		}
+		a.Properties.OAuth2 = authProperties
 		return nil
-	}
-	if err := unmarshalKey("metadata", properties, &b.Metadata); err != nil {
-		return err
-	}
-	if err := unmarshalKey("secret", properties, &b.Secret); err != nil {
-		return err
-	}
-	return nil
-}
 
-// GetMetadata ...
-func (b *BaseAuthProperties) GetMetadata() *Metadata {
-	return &b.Metadata
+	default:
+		return fmt.Errorf("failed to parse auth properties")
+	}
 }
 
-// GetSecret ...
-func (b *BaseAuthProperties) GetSecret() string {
-	return b.Secret
+func (a *Auth) MarshalJSON() ([]byte, error) {
```

Comment:
The current test covers this method?

Reply:
there are a few tests that covers it, e.g.:
https://github.com/serverlessworkflow/sdk-go/blob/main/parser/parser_test.go#L663
https://github.com/serverlessworkflow/sdk-go/blob/main/parser/parser_test.go#L600